modules/library/main/src/test/java/org/geotools/feature/visitor/GroupByVisitorTest.java

```diff
@@ -305,14 +329,21 @@ private void testVisitor(String aggregateAttribute, String aggregateVisitor,
      */
     private GroupByVisitor executeVisitor(FeatureCollection featureCollection, String aggregateAttribute,
                                           String aggregateVisitor, String[] groupByAttributes) throws Exception {
+        GroupByVisitor visitor = buildVisitor(aggregateAttribute, aggregateVisitor,
+                groupByAttributes);
+        featureCollection.accepts(visitor, new NullProgressListener());
+        return visitor;
+    }
+
+    private GroupByVisitor buildVisitor(String aggregateAttribute, String aggregateVisitor,
+            String[] groupByAttributes) {
         GroupByVisitorBuilder visitorBuilder = new GroupByVisitorBuilder()
                 .withAggregateAttribute(aggregateAttribute, buildingType)
                 .withAggregateVisitor(aggregateVisitor);
         for (String groupByAttribute : groupByAttributes) {
             visitorBuilder.withGroupByAttribute(groupByAttribute, buildingType);
         }
         GroupByVisitor visitor = visitorBuilder.build();
-        featureCollection.accepts(visitor, new NullProgressListener());
         return visitor;
```

Comment:
We often do this on purpose to simplify debugging (gives a place where the variable can be examined before exiting the method), but sure, in this case it adds no value